File: PetrolStation.View/Responses/PetrolStationDetailsResponse.cs
using System;
using System.Collections.Generic;

namespace PetrolStation.View.Responses
{
    public class PetrolStationDetailsResponse
    {
        public PetrolStationDetailsResponse(Guid id, string name, DateTime time, IReadOnlyCollection<TankResponse> tanks, IReadOnlyCollection<PetrolStationRaportResponse> raports)
        {
            Id = id;
            Name = name;
            Time = time;
            Tanks = tanks;
            Raports = raports;
        }

        public Guid Id { get; }

        public string Name { get; }

        public DateTime Time { get; }

        public IReadOnlyCollection<TankResponse> Tanks { get; }
        public IReadOnlyCollection<PetrolStationRaportResponse> Raports { get; }
    }
}

File: PetrolStation.View/ClientApp/src/app/models/PetrolStationDetailsResponse.ts
interface NozzleResponse {
    id: string;
    lastTransactionVolume: number;
    totalStolenPetrolAmount: number;
    lastStolenPetrolAmount: number;
    totalPetrolAmount: number;
}

interface TankResponse {
    id: string;
    maximumVolume: number;
    currentVolume: number;
    petrolTemperature: number;
    tankHigh: number;
    nozzleResponses: NozzleResponse[];
}

interface PetrolStationRaportResponse {
    tankId: string;
    nozzleId: string;
    timeStamp: Date;
    report: string;
    status: string;
}

interface PetrolStationDetailsResponse {
    id: string;
    name: string;
    time: Date;
    tanks: TankResponse[];
    raports: PetrolStationRaportResponse[];
}
